ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -352,36 +354,44 @@ showTracers tr = Tracers {
   Applications
 -------------------------------------------------------------------------------}
 
+-- | A node-to-node application
+type App m peer bytes a =
+     NodeToNodeVersion
+  -> ControlMessageSTM m
+  -> peer
+  -> Channel m bytes
+  -> m (a, Maybe bytes)
+
 -- | Applications for the node-to-node protocols
 --
 -- See 'Network.Mux.Types.MuxApplication'
 data Apps m peer bCS bBF bTX bKA a = Apps {
       -- | Start a chain sync client that communicates with the given upstream
       -- node.
-      aChainSyncClient    :: NodeToNodeVersion -> peer -> Channel m bCS -> m (a, Maybe bCS)
+      aChainSyncClient    :: App m peer bCS a
```

Comment:
This means we'll not see the actual return value of the client, even though all the upstream code changes now make it available. (The `a` is shared by the other protocols, and none of those return anything. So it's `()`/`Void`/`void`/etc or at the very least a bit misleading.) The CS return value is recorded in a tracer invoked within the client, so it'll at least be logged, which seems like the primary use-case, though.

Just an observation.

Reply:
Yeah, that's indeed not so nice. That `Apps` business is actually written by the network team. Unfortunately it lives in the consensus layer :grimacing:.